src/Application/Regulation/Query/GetRegulationOrdersToDatexFormatQueryHandler.php

```diff
@@ -18,36 +19,60 @@ public function __construct(
 
     public function __invoke(GetRegulationOrdersToDatexFormatQuery $query): array
     {
-        $regulationOrders = $this->repository->findRegulationOrdersForDatexFormat();
+        $rows = $this->repository->findRegulationOrdersForDatexFormat();
+
+        // There is one row per unique combination of (regulationOrder, location, measure).
+        // Rows are sorted by regulationOrder uuid.
+        // So we iterate over rows, pushing a new regulation order view when the row's regulationOrder uuid changes.
         $regulationOrderViews = [];
+        $index = 0;
 
-        foreach ($regulationOrders as $regulationOrder) {
-            $vehicleConditions = [];
+        while ($index < \count($rows)) {
+            $currentRegulationOrder = $rows[$index];
+            $trafficRegulations = [];
 
-            foreach ($regulationOrder['restrictedVehicleTypes'] ?: [] as $restrictedVehicleType) {
-                $vehicleConditions[] = new DatexVehicleConditionView($restrictedVehicleType);
-            }
+            while ($index < \count($rows)) {
```

Comment:
Deux petites remarques : 
- Dans un souci de performance, ne vaudrait-il pas mieux sortir `\count($rows)` dans une variable à l'extérieur des boucles pour éviter de recalculer l'opération à chaque itération ?
- Je ne comprends pas trop l'intérêt de faire deux boucles `while` ici. Il y a peut-être une subtilité que je n'ai pas saisie…

Reply:
J'ai réécrit le parcours de façon différente, avec un simple `foreach`

La subtilité c'est qu'il faut bien penser à "flush" la view finale, ce qui n'est pas le plus élégant puisqu'il y a un peu de duplication de code. Mais pas trop de risque qu'on ait à modifier l'un sans devoir modifier l'autre, je pense... Si tu as une idée pour améliorer, je suis preneur cela dit.